tests/ReflectionTypeTest.php

```diff
@@ -85,14 +72,31 @@ public function testTypeConvertToDisplayTypeImplicitlyNullable()
         $this->assertEquals(\ReflectionParameter::class, get_class($nativeParamRefArr[0]));
         $nativeTypeRef = $nativeParamRefArr[0]->getType();
         $this->assertTrue($nativeTypeRef->allowsNull());
-        // TODO: To prevent deprecation error in tests
-        if (PHP_VERSION_ID < 70400) {
-            $this->assertEquals('string', (string)$nativeTypeRef);
-        } else {
-            $this->assertEquals('string', $nativeTypeRef->getName());
-        }
+        $this->assertEquals('string', $nativeTypeRef->getName());
         $this->assertStringNotContainsString('\\', get_class($nativeTypeRef));
         $this->assertInstanceOf(\ReflectionType::class, $nativeTypeRef);
-        $this->assertEquals('string or NULL', \Go\ParserReflection\ReflectionType::convertToDisplayType($nativeTypeRef));
+        $this->assertEquals('?string', \Go\ParserReflection\ReflectionType::convertToDisplayType($nativeTypeRef));
+    }
+
+    /**
+     * Testing convertToDisplayType() with native \ReflectionType
+     *
+     * We're already testing it with Go\ParserReflection\ReflectionType
+     * elsewhere.
+     */
+    public function testTypeConvertToDisplayTypeImplicitlyUnionNullable()
+    {
+        $nativeClassRef = new \ReflectionClass('Go\\ParserReflection\\Stub\\ClassWithPhp80Features');
+        $nativeMethodRef = $nativeClassRef->getMethod('acceptsStringArrayDefaultToNull');
+        $this->assertEquals(\ReflectionMethod::class, get_class($nativeMethodRef));
+        $nativeParamRefArr = $nativeMethodRef->getParameters();
+        $this->assertCount(1, $nativeParamRefArr);
+        $this->assertEquals(\ReflectionParameter::class, get_class($nativeParamRefArr[0]));
+        $nativeTypeRef = $nativeParamRefArr[0]->getType();
+        $this->assertTrue($nativeTypeRef->allowsNull());
+        $this->assertEquals('array|string|null', $nativeTypeRef);
```

Comment:
there is no `->getName()` on native `ReflectionUnionType`, so it just use its `__toString()` convert, got full type `array|string|null`